### ISSUE TITLE
Bug: Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by daviddelusenet** [IBM/audit-ci#176]
Hello,

I've the following NPM script in my `package.json`:

```bash
"audit": "audit-ci --moderate"
```

When I run this script by running `npm run audit` I get the following error:

```bash
audit-ci version: 3.1.1
Cannot convert undefined or null to object
Exiting...
```

NPM version: 7.6.3
Node version: 12.18.2

Please let me know if you need any more additional information.

**Comment by daviddelusenet** [comment]
I downgraded my NPM version to `6.14.11` and everything works fine again..

**Comment by quinnturner** [comment]
We are adding NPM v7 support (hopefully today) in this PR: https://github.com/IBM/audit-ci/pull/175 😄

**Comment by daviddelusenet** [comment]
I'm experiencing this issue again, but now when using NPM version `6.14.11`.

```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/Users/daviddelusenet/.nvm/versions/node/v12.18.2/bin/node',
1 verbose cli   '/Users/daviddelusenet/.nvm/versions/node/v12.18.2/bin/npm',
1 verbose cli   'run',
1 verbose cli   'audit'
1 verbose cli ]
2 info using npm@6.14.11
3 info using node@v12.18.2
4 verbose run-script [ 'preaudit', 'audit', 'postaudit' ]
5 info lifecycle @codedazur/react-components@4.8.1~preaudit: @codedazur/react-components@4.8.1
6 info lifecycle @codedazur/react-components@4.8.1~audit: @codedazur/react-components@4.8.1
7 verbose lifecycle @codedazur/react-components@4.8.1~audit: unsafe-perm in lifecycle true
8 verbose lifecycle @codedazur/react-components@4.8.1~audit: PATH: /Users/daviddelusenet/.nvm/versions/node/v12.18.2/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/daviddelusenet/Development/react-components/node_modules/.bin:/usr/local/opt/php@7.4/sbin:/usr/local/opt/php@7.4/bin:/Users/daviddelusenet/.nvm/versions/node/v12.18.2/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
9 verbose lifecycle @codedazur/react-components@4.8.1~audit: CWD: /Users/daviddelusenet/Development/react-components
10 silly lifecycle @codedazur/react-components@4.8.1~audit: Args: [ '-c', 'audit-ci' ]
11 silly lifecycle @codedazur/react-components@4.8.1~audit: Returned: code: 1  signal: null
12 info lifecycle @codedazur/react-components@4.8.1~audit: Failed to exec audit script
13 verbose stack Error: @codedazur/react-components@4.8.1 audit: `audit-ci`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/Users/daviddelusenet/.nvm/versions/node/v12.18.2/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:315:20)
13 verbose stack     at ChildProcess.<anonymous> (/Users/daviddelusenet/.nvm/versions/node/v12.18.2/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:315:20)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
14 verbose pkgid @codedazur/react-components@4.8.1
15 verbose cwd /Users/daviddelusenet/Development/react-components
16 verbose Darwin 20.3.0
17 verbose argv "/Users/daviddelusenet/.nvm/versions/node/v12.18.2/bin/node" "/Users/daviddelusenet/.nvm/versions/node/v12.18.2/bin/npm" "run" "audit"
18 verbose node v12.18.2
19 verbose npm  v6.14.11
20 error code ELIFECYCLE
21 error errno 1
22 error @codedazur/react-components@4.8.1 audit: `audit-ci`
22 error Exit status 1
23 error Failed at the @codedazur/react-components@4.8.1 audit script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by daviddelusenet** [comment]
@quinnturner can this issue be reopened? It has to do with the update of some dependency, this is the result of `npm outdated` in my project:

```bash
@babel/core                                 7.10.2  7.13.14  7.13.14  @codedazur/react-components
@babel/plugin-proposal-class-properties     7.10.1   7.13.0   7.13.0  @codedazur/react-components
@babel/plugin-proposal-object-rest-spread   7.10.1   7.13.8   7.13.8  @codedazur/react-components
@babel/preset-env                           7.10.2  7.13.12  7.13.12  @codedazur/react-components
@babel/preset-react                         7.10.1  7.13.13  7.13.13  @codedazur/react-components
@babel/preset-typescript                    7.10.1   7.13.0   7.13.0  @codedazur/react-components
@commitlint/cli                              8.3.5    8.3.5   12.1.1  @codedazur/react-components
@commitlint/config-conventional              8.3.4    8.3.4   12.1.1  @codedazur/react-components
@semantic-release/npm                        7.0.9    7.1.0    7.1.0  @codedazur/react-components
@semantic-release/release-notes-generator    9.0.1    9.0.2    9.0.2  @codedazur/react-components
@storybook/addon-a11y                       5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/addon-actions                    5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/addon-docs                       5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/addon-knobs                      5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/addon-links                      5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/addon-storyshots                 5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/addon-storysource                5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/addon-viewport                   5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/core                             5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/react                            5.3.19   5.3.21    6.2.3  @codedazur/react-components
@storybook/source-loader                    5.3.19   5.3.21    6.2.3  @codedazur/react-components
@types/jest                                 25.2.3   25.2.3  26.0.22  @codedazur/react-components
@types/react                               16.9.35  16.14.5   17.0.3  @codedazur/react-components
@types/react-dom                            16.9.8  16.9.12   17.0.3  @codedazur/react-components
@types/react-helmet                         5.0.16   5.0.16    6.1.0  @codedazur/react-components
@types/react-test-renderer                  16.9.2   16.9.5   17.0.1  @codedazur/react-components
@types/styled-components                     5.1.0    5.1.9    5.1.9  @codedazur/react-components
@types/uuid                                  7.0.4    7.0.4    8.3.0  @codedazur/react-components
@types/webpack                             4.41.17  4.41.27   5.28.0  @codedazur/react-components
@typescript-eslint/eslint-plugin            2.34.0   2.34.0   4.21.0  @codedazur/react-components
@typescript-eslint/parser                   2.34.0   2.34.0   4.21.0  @codedazur/react-components
audit-ci                                     2.5.1    2.5.1    3.1.1  @codedazur/react-components
babel-jest                                  25.5.1   25.5.1   26.6.3  @codedazur/react-components
babel-loader                                 8.1.0    8.2.2    8.2.2  @codedazur/react-components
babel-plugin-styled-components              1.10.7   1.12.0   1.12.0  @codedazur/react-components
commitizen                                   4.2.1    4.2.3    4.2.3  @codedazur/react-components
cz-conventional-changelog                    3.2.0    3.3.0    3.3.0  @codedazur/react-components
eslint                                       6.8.0    6.8.0   7.23.0  @codedazur/react-components
eslint-config-prettier                      6.11.0   6.15.0    8.1.0  @codedazur/react-components
eslint-plugin-prettier                       3.1.3    3.3.1    3.3.1  @codedazur/react-components
eslint-plugin-react                         7.20.0   7.23.1   7.23.1  @codedazur/react-components
eslint-plugin-react-hooks                    3.0.0    3.0.0    4.2.0  @codedazur/react-components
html-react-parser                           0.10.5   0.10.5    1.2.4  @codedazur/react-components
husky                                        4.2.5    4.3.8    6.0.0  @codedazur/react-components
jest                                        25.5.4   25.5.4   26.6.3  @codedazur/react-components
jest-styled-components                       7.0.2    7.0.3    7.0.3  @codedazur/react-components
prettier                                     2.0.5    2.2.1    2.2.1  @codedazur/react-components
react                                      16.13.1  16.14.0   17.0.2  @codedazur/react-components
react-dom                                  16.13.1  16.14.0   17.0.2  @codedazur/react-components
react-is                                   16.13.1  16.13.1   17.0.2  @codedazur/react-components
react-spring                                8.0.27   8.0.27    9.0.0  @codedazur/react-components
react-test-renderer                        16.13.1  16.14.0   17.0.2  @codedazur/react-components
semantic-release                            17.3.0   17.4.2   17.4.2  @codedazur/react-components
serverless                                  1.83.0   1.83.3   2.33.1  @codedazur/react-components
serverless-s3-deploy                         0.9.0    0.9.0   0.10.0  @codedazur/react-components
styled-components                            5.1.1    5.2.3    5.2.3  @codedazur/react-components
typescript                                   3.9.5    3.9.9    4.2.3  @codedazur/react-components
uuid                                         8.1.0    8.3.2    8.3.2  @codedazur/react-components
webpack                                     4.44.1   4.46.0   5.30.0  @codedazur/react-components
webpack-cli                                 3.3.11   3.3.12    4.6.0  @codedazur/react-components
```

When I run `npm update` the error starts to appear.

**Comment by daviddelusenet** [comment]
I've managed to pinpoint the issue, it's because of the following package update:

```bash
@semantic-release/npm                        7.0.9    7.1.0    7.1.0  @codedazur/react-components
```

Is this a problem on their end of a bug in `audit-ci`? Either way, I'll open an issue in their Github also.

**Comment by quinnturner** [comment]
<img width="692" alt="image" src="https://user-images.githubusercontent.com/12538019/113862513-de871400-977e-11eb-8e72-5e8627048da0.png">
Here's my attempt at reproduction, which didn't reproduce. I tried installing `7.0.9`, then upgrading to `7.1.0`. Both resulted in zero results found when performing `npm audit` as well as `audit-ci`.
`@semantic-release/npm` seems to have quite a few transitive dependencies. Can you perform a regular `npm audit` (or `npm audit --json` and show the results? Or, find the diff in `package-lock.json` when updating `@semantic-release/npm` to isolate the underlying NPM issue.

**Comment by quinnturner** [comment]
I kind of think this is related to NPM v7 still since I got a similar issue trying on v7. Note that we have a PR that's almost ready to ship for NPM v7 here #175. Perhaps let's try a reproduction after that gets merged? (hopefully today!)

**Comment by daviddelusenet** [comment]
The diff when updating from `7.0.9` from `7.1.0` is huge but I think this might be the issue right?

```json
     "npm": {
-      "version": "6.14.12",
-      "resolved": "https://registry.npmjs.org/npm/-/npm-6.14.12.tgz",
-      "integrity": "sha512-La0TNNm1TLYaSeOyit+p3xGTRYYRsHae6/RG69MVXurZsWna9jccPP7FOi/u7V9WdiCV5OOojrfMD+WstO5MZQ==",
+      "version": "7.8.0",
+      "resolved": "https://registry.npmjs.org/npm/-/npm-7.8.0.tgz",
+      "integrity": "sha512-9AC3Dj9OUWaUdmTmEVttE/1MWkfF7+sAKPRo9tKEyjo49AXmHQBn+RC33M9dima91mEMqDIA71xyRm4VmhDipg==",
```

**Comment by quinnturner** [comment]
That is definitely the issue, 100%. NPM 7 support is probably gonna get merged by today or tonight, so this issue should go away soon :)

**Comment by menocomp** [comment]
Thanks for sharing such a nice library publicly. I appreciate your work.
Currently I am using this version: `3.2.0`, Should I expect to get that fix in the next tag?

**Comment by quinnturner** [comment]
I just merged support for NPM 7 into `master`. Hopefully, it will be released today in `v4.0.0`. Sorry for the delay everyone!

**Comment by quinnturner** [comment]
I am closing this until further notice as there's not much else to do from `audit-ci`'s perspective now that NPM 7 is supported. Thanks everyone!
